app/(signed-in)/bookings/page.tsx

```diff
@@ -1,7 +1,97 @@
-import React from "react";
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { useSession } from "next-auth/react";
+import {
+  PartialReservation,
+  retrieveAllReservations,
+} from "@/server-actions/reservation-actions";
+import Link from "next/link";
 
 const BookingsPage = () => {
-  return <div>BookingsPage</div>;
+  const { data: session, status } = useSession();
+  const [reservations, setReservations] = useState<PartialReservation[] | []>(
+    []
+  );
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
```

Comment:
nit: if this should only be ran one time (not on each rerender), use `use`. `use` also integrates nicely with async loading of necessary data to render. While the data is awaited, a suspense boundary is hit. This is a good way to avoid `loading` state's.

Reply:
Made these SSR